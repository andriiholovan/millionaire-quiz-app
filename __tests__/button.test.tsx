import {
  afterEach,
  describe,
  expect,
  it,
  vi,
} from 'vitest';
import {
  fireEvent,
  cleanup,
  render,
  screen,
} from '@testing-library/react';
import Button from '@/app/_components/button';

describe('BUTTON COMPONENT', () => {
  afterEach(() => {
    cleanup();
    vi.restoreAllMocks();
  });

  it('Should match snapshot of <button> element with default props only', () => {
    const { asFragment } = render(
      <Button>Start</Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot of <a> element', () => {
    const { asFragment } = render(
      <Button push to="/">Start</Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot of Button.Primary component', () => {
    const { asFragment } = render(
      <Button.Primary push to="/">Start</Button.Primary>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot of Button.Icon component', () => {
    const { asFragment } = render(
      <Button.Icon
        iconAlt="Alt text"
        onClick={vi.fn()}
        src="/path/to/any/icon"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should execute onClick handler in Button.Icon', () => {
    const mockFn = vi.fn();
    render(
      <Button.Icon
        iconAlt="Menu button"
        onClick={mockFn}
        src="/path/to/any/icon"
      />,
    );
    fireEvent.click(screen.getByRole('button'));
    expect(mockFn).toHaveBeenCalledOnce();
  });
});
