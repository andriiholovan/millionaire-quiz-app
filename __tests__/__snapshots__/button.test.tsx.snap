// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BUTTON COMPONENT > Should match snapshot of <a> element 1`] = `
<DocumentFragment>
  <a
    class="_button_6961ec"
    href="/"
  >
    Start
  </a>
</DocumentFragment>
`;

exports[`BUTTON COMPONENT > Should match snapshot of <button> element with default props only 1`] = `
<DocumentFragment>
  <button
    class="_button_6961ec"
    type="button"
  >
    Start
  </button>
</DocumentFragment>
`;

exports[`BUTTON COMPONENT > Should match snapshot of Button.Icon component 1`] = `
<DocumentFragment>
  <button
    class="_button_6961ec _button_mobile_6961ec _button_align_right_6961ec"
    type="button"
  >
    <img
      alt="Alt text"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="24"
      src="/_next/image?url=%2Fpath%2Fto%2Fany%2Ficon&w=48&q=75"
      srcset="/_next/image?url=%2Fpath%2Fto%2Fany%2Ficon&w=32&q=75 1x, /_next/image?url=%2Fpath%2Fto%2Fany%2Ficon&w=48&q=75 2x"
      style="color: transparent;"
      width="24"
    />
  </button>
</DocumentFragment>
`;

exports[`BUTTON COMPONENT > Should match snapshot of Button.Primary component 1`] = `
<DocumentFragment>
  <a
    class="_button_6961ec _button_wide_6961ec _button_primary_6961ec"
    href="/"
  >
    Start
  </a>
</DocumentFragment>
`;
